ports/mimxrt10xx/common-hal/microcontroller/Pin.c

```diff
@@ -31,170 +31,69 @@
 
-//STATIC uint32_t never_reset_pins[PORT_COUNT];
+// GPIO ports are 32 pins wide
+STATIC uint32_t claimed_pins[GPIO_PORT_COUNT]; //remove?
+STATIC uint32_t never_reset_pins[GPIO_PORT_COUNT];
 
+// There are two numbering systems used here:
+// IOMUXC index, used for iterating through pins and accessing reset information, 
+// and GPIO port and number, used to store claimed and reset tagging. The two number 
+// systems are not related and one cannot determine the other without a pin object
 void reset_all_pins(void) {
-//    uint32_t pin_mask[PORT_COUNT] = PORT_OUT_IMPLEMENTED;
-
-//    // Do not full reset USB lines.
-//    pin_mask[0] &= ~(PORT_PA24 | PORT_PA25);
-
-//    // Do not reset SWD when a debugger is present.
-//    if (DSU->STATUSB.bit.DBGPRES == 1) {
-//        pin_mask[0] &= ~(PORT_PA30 | PORT_PA31);
-//    }
-
-//    for (uint32_t i = 0; i < PORT_COUNT; i++) {
-//        pin_mask[i] &= ~never_reset_pins[i];
-//    }
-
-//    gpio_set_port_direction(GPIO_PORTA, pin_mask[0] & ~MICROPY_PORT_A, GPIO_DIRECTION_OFF);
-//    gpio_set_port_direction(GPIO_PORTB, pin_mask[1] & ~MICROPY_PORT_B, GPIO_DIRECTION_OFF);
-//    #if PORT_BITS > 64
-//    gpio_set_port_direction(GPIO_PORTC, pin_mask[2] & ~MICROPY_PORT_C, GPIO_DIRECTION_OFF);
-//    #endif
-//    #if PORT_BITS > 96
-//    gpio_set_port_direction(GPIO_PORTD, pin_mask[3] & ~MICROPY_PORT_D, GPIO_DIRECTION_OFF);
-//    #endif
-//
-//    // Configure SWD. SWDIO will be automatically switched on PA31 when a signal is input on
-//    // SWCLK.
-//    #ifdef SAMD51
-//    gpio_set_pin_function(PIN_PA30, MUX_PA30H_CM4_SWCLK);
-//    #endif
-//    #ifdef SAMD21
-//    gpio_set_pin_function(PIN_PA30, GPIO_PIN_FUNCTION_G);
-//    gpio_set_pin_function(PIN_PA31, GPIO_PIN_FUNCTION_G);
-//    #endif
-//
+    for (uint8_t i = 0; i < GPIO_PORT_COUNT; i++) {
+        claimed_pins[i] = never_reset_pins[i];
+    }
+    for (uint8_t i = 0; i < IOMUXC_SW_PAD_CTL_PAD_COUNT; i++) {
+        // mp_printf(&mp_plat_print, "CTL MUX i=%d:%x\n",i,IOMUXC->SW_MUX_CTL_PAD[i]);
+        // mp_printf(&mp_plat_print, "CTL PAD i=%d:%x\n",i,IOMUXC->SW_PAD_CTL_PAD[i]);
+        if(!(never_reset_pins[mcu_pin_list[i].port] & (1 << mcu_pin_list[i].number))) {
+            // mp_printf(&mp_plat_print, "SUCCESS\n");
+            IOMUXC->SW_MUX_CTL_PAD[i] = mcu_pin_list[i].mux_reset;
+            IOMUXC->SW_PAD_CTL_PAD[i] = mcu_pin_list[i].pad_reset;
+        }
+    }
+
     #ifdef MICROPY_HW_NEOPIXEL
     neopixel_in_use = false;
     #endif
-    #ifdef MICROPY_HW_APA102_MOSI
-    apa102_sck_in_use = false;
-    apa102_mosi_in_use = false;
-    #endif
-
-    // After configuring SWD because it may be shared.
-    #ifdef SPEAKER_ENABLE_PIN
-    speaker_enable_in_use = false;
-//    gpio_set_pin_function(SPEAKER_ENABLE_PIN->number, GPIO_PIN_FUNCTION_OFF);
-//    gpio_set_pin_direction(SPEAKER_ENABLE_PIN->number, GPIO_DIRECTION_OUT);
-//    gpio_set_pin_level(SPEAKER_ENABLE_PIN->number, false);
-    #endif
```

Comment:
Not sure these should be removed, there might be boards that will have a APA102 and/or SPEAKER and you want to keep track of.

Reply:
I personally don't feel it's a good pattern to imply functionality that doesn't actually exist in the code, like Speaker pins for a board that doesn't have AudioIO yet. I'd rather just bring in the functionality from other ports at the time that capability is being coded in. 

That said I thought an APA102 was AudioIO related and should have done a quick google search sooner to discover otherwise. I'll turn it back on and enable it for the STM32 also as soon as I can.

Reply:
Ok, then maybe for each removed TODO in code we create an Issue so we don't forget what is left to be done :)